src/overlays/actors/ovl_En_Peehat/z_en_peehat.c

```diff
@@ -92,85 +99,992 @@ static ColliderJntSphInit D_80AD2790 = {
+            thisx->naviEnemyId = 0x49; // Larva
+            EnPeehat_Larva_SetStateSeekPlayer(this);
+    }
+}
+
+void EnPeehat_Destroy(Actor* thisx, GlobalContext* globalCtx) {
+    EnPeehat* this = THIS;
+    EnPeehat* parent;
+
+    Collider_DestroyCylinder(globalCtx, &this->colCylinder);
+    Collider_DestroyJntSph(globalCtx, &this->colJntSph);
+
+    // If larva, decrement total larva spawned
+    if (thisx->params > 0) {
+        parent = (EnPeehat*)thisx->parent;
+        if (parent != NULL && parent->actor.update != NULL) {
+            parent->unk2FA--;
+        }
+    }
+}
+
+void EnPeehat_SpawnDust(GlobalContext* globalCtx, EnPeehat* this, Vec3f* pos, f32 arg3, s32 arg4, f32 arg5, f32 arg6) {
+    Vec3f pPos;
+    Vec3f pVel = { 0, 8, 0 };
+    Vec3f pAccel = { 0, -1.5f, 0 };
+    f32 rot; // radians
+    s32 pScale;
+
+    rot = (Rand_ZeroOne() - 0.5f) * 6.28f;
+    pPos.y = this->actor.floorHeight;
+    pPos.x = (Math_SinF(rot) * arg3) + pos->x;
+    pPos.z = (Math_CosF(rot) * arg3) + pos->z;
+    pAccel.x = (Rand_ZeroOne() - 0.5f) * arg5;
+    pAccel.z = (Rand_ZeroOne() - 0.5f) * arg5;
+    pVel.y += ((Rand_ZeroOne() - 0.5f) * 4.0f);
+    pScale = (s32)(((Rand_ZeroOne() * 5.0f) + 12.0f) * arg6);
+    EffectSsHahen_Spawn(globalCtx, &pPos, &pVel, &pAccel, arg4, pScale, -1, 10, NULL);
+}
+
+/*
+ * Handles being hit when on the ground
+ */
+void EnPeehat_HitWhenGrounded(EnPeehat* this, GlobalContext* globalCtx) {
+    this->colCylinder.base.acFlags &= ~AC_HIT;
+    if ((globalCtx->gameplayFrames & 0xF) == 0) {
+        Vec3f itemDropPos = this->actor.world.pos;
+        itemDropPos.y += 70.0f;
+        Item_DropCollectibleRandom(globalCtx, &this->actor, &itemDropPos, 0x40);
+        Item_DropCollectibleRandom(globalCtx, &this->actor, &itemDropPos, 0x40);
+        Item_DropCollectibleRandom(globalCtx, &this->actor, &itemDropPos, 0x40);
+        this->unk2D4 = 240;
+    } else {
+        s32 i;
+        this->colCylinder.base.acFlags &= ~AC_HIT;
+        for (i = MAX_LARVA - this->unk2FA; i > 0; i--) {
+            Actor* larva = Actor_SpawnAsChild(&globalCtx->actorCtx, &this->actor, globalCtx, ACTOR_EN_PEEHAT,
+                                              Rand_CenteredFloat(25.0f) + this->actor.world.pos.x,
+                                              Rand_CenteredFloat(25.0f) + (this->actor.world.pos.y + 50.0f),
+                                              Rand_CenteredFloat(25.0f) + this->actor.world.pos.z, 0, 0, 0, TYPE_LARVA);
+            if (larva != NULL) {
+                larva->velocity.y = 6.0f;
+                larva->shape.rot.y = larva->world.rot.y = (s16)Rand_CenteredFloat(0xFFFF);
+                this->unk2FA++;
+            }
+        }
+        this->unk2D4 = 8;
+    }
+    Audio_PlayActorSound2(&this->actor, NA_SE_EN_PIHAT_DAMAGE);
+}
+
+void EnPeehat_Ground_SetStateGround(EnPeehat* this) {
+    Animation_Change(&this->skelAnime, &D_060009C4, 0.0f, 3.0f, Animation_GetLastFrame(&D_060009C4.common), 2, 0.0f);
```

Comment:
```suggestion
    Animation_Change(&this->skelAnime, &D_060009C4, 0.0f, 3.0f, Animation_GetLastFrame(&D_060009C4), 2, 0.0f);
```
`Animation_GetLastFrame` takes `void*`, so you don't need this. The same applies to the other animations.